R/uvhydrograph-render.R

```diff
@@ -353,18 +367,25 @@ createSecondaryPlot <- function(uvInfo, secondarySeriesList,
         timezone, getSecondaryPlotConfig, list(uvInfo[['label']], lims$ylim), excludeZeroNegativeFlag)
   }
   
-  #estimated data
-  if(!isEmptyOrBlank(secondarySeriesList[['estimated']]) && !isEmptyVar(secondarySeriesList[['estimated']][['points']])) {
-    plot_object <- plotTimeSeries(plot_object, secondarySeriesList[['estimated']], "estimated", 
-        timezone, getSecondaryPlotConfig, list(uvInfo[['label']], lims$ylim), excludeZeroNegativeFlag)
-  }
-  
   #corrected data
   # primary axis is logged based on corrected data; if no corrected data, should be FALSE
   # gsplot object logged inside of plotTimeSeries >> plotItem
-  doLog <- isLogged(secondarySeriesList[['corrected']][['points']], secondarySeriesList[['corrected']][['isVolumetricFlow']], excludeZeroNegativeFlag)
-  plot_object <- plotTimeSeries(plot_object, secondarySeriesList[['corrected']], "corrected", 
+  if (secondarySeriesList[['useEstimated']]) {
+  # Plot estimated corrected data, in place of non-estimated corrected data
+    doLog <- isLogged(secondarySeriesList[['estimated']][['points']], secondarySeriesList[['estimated']][['isVolumetricFlow']], excludeZeroNegativeFlag)
+    plot_object <- plotTimeSeries(plot_object, secondarySeriesList[['estimated']], "estimated", 
       timezone, getSecondaryPlotConfig, list(uvInfo[['label']], lims$ylim), excludeZeroNegativeFlag)
+  } else {
+  	#Plot estimated data if non-estimated corrected data exists
+  	if(!isEmptyOrBlank(secondarySeriesList[['estimated']]) && !isEmptyVar(secondarySeriesList[['estimated']][['points']])) {
```

Comment:
so this `plotTimeSeries` call is happening for `secondarySeriesList[['estimated']]` no matter what. Seems like it can remain outside of the if-statement like it was before?